src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/config/EmailConfiguration.kt

```diff
@@ -60,19 +45,29 @@ fun interface EmailService {
   fun send(email: Email): Result<UUID>
 }
 
-abstract class Email {
-  abstract val address: String
-  abstract val personalisation: Map<String, String?>
+abstract class Email(
+  val address: String,
+  prisonerFirstName: String,
+  prisonerLastName: String,
+  prisonerNumber: String,
+  date: LocalDate = LocalDate.now(),
+  comments: String? = "",
+) {
+  private val common = mapOf(
+    "date" to date.toString(),
+    "prisonerName" to prisonerFirstName.plus(" $prisonerLastName"),
```

Comment:
Something to keep an eye out for.  Some templates have just 'prisonerName' whilst other have 'prisonerfirstName' and 'prisonerLastName' attributes.

Reply:
We can standardise the approach as we come across differences.